lib/core/usecase_models/extend_research_context_usecase_models.py

```diff
@@ -0,0 +1,47 @@
+from typing import List
+from pydantic import Field
+from lib.core.entity.models import LLM, ResearchContext
+from lib.core.sdk.usecase_models import BaseErrorResponse, BaseRequest, BaseResponse
+
+
+class ExtendResearchContextRequest(BaseRequest):
+    """
+    Request Model for the Use Case where a Research Context is Extended beyond its original data sources.
+
+    @param new_research_context_title: Title of the new research context to be created to include new data sources.
+    @param client_sub: SUB of the client for which the research context is to be created.
+    @param llm_name: Name of the LLM for which the research context is to be created.
+    @param new_source_data_ids: List of additional source data ids beyond those in the original research context.
+    @param existing_research_context_id: ID of the existing research context to be extended.
+    """
+
+    new_research_context_title: str = Field(
+        description="Title of the new research context to be created to include new data sources."
+    )
+    new_research_context_description: str = Field(description="Description of the research context to be created.")
+    client_sub: str = Field(description="SUB of the client for which the research context is to be created.")
```

Comment:
@ydamit There's a repository function in sqla_research_context_repository called `get_research_context_client`. You can use this one, instead of asking for a `client_sub` as an input. If you do this, you'll directly get the `client` you need and you wouldn't need to check if the `client` provided as an input matches the `client` of the old research_context

Reply:
Noted! But since we need `client_sub` to create _new_ research contexts, I just want to triple-check: we never want to allow one SUB to extend a research context originally created by another SUB?

Reply:
Actually, since we're at it, you're right, we might want this eventually. If we implement this, then we'll change your check for some sort of permission check. So no problem, please just leave this as is!